pysmt/walkers/identitydag.py

```diff
@@ -186,6 +189,39 @@ def walk_bv_sdiv(self, formula, args, **kwargs):
     def walk_bv_srem(self, formula, args, **kwargs):
         return self.mgr.BVSRem(args[0], args[1])
 
+    def walk_str_length(self, formula, args, **kwargs):
```

Comment:
Where is the association of this walk function to the corresponding node type? I cannot find it.

Reply:
This is done in walkers/generic.py:26 . The association is by default the lower-case version of the node name.